sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -443,6 +443,10 @@ case class RangeExec(range: org.apache.spark.sql.catalyst.plans.logical.Range)
       |     if (shouldStop()) return;
       |   }
       |
+      |   if (TaskContext.get().isInterrupted()) {
```

Comment:
You could also move TaskContext.get() out of the loop. It should not change during evaluation.